pkg/morph/client/container/wrapper/container.go

```diff
@@ -1,38 +1,123 @@
 package wrapper
 
-// OwnerID represents the container owner identifier.
-// FIXME: correct the definition.
-type OwnerID struct{}
+import (
+	"crypto/sha256"
 
-// Container represents the NeoFS Container structure.
-// FIXME: correct the definition.
-type Container struct{}
+	"github.com/nspcc-dev/neofs-api-go/pkg/container"
+	"github.com/nspcc-dev/neofs-api-go/pkg/owner"
+	v2container "github.com/nspcc-dev/neofs-api-go/v2/container"
+	msgContainer "github.com/nspcc-dev/neofs-api-go/v2/container/grpc"
+	v2refs "github.com/nspcc-dev/neofs-api-go/v2/refs"
+	msgRefs "github.com/nspcc-dev/neofs-api-go/v2/refs/grpc"
+	core "github.com/nspcc-dev/neofs-node/pkg/core/container"
+	client "github.com/nspcc-dev/neofs-node/pkg/morph/client/container"
+	"github.com/pkg/errors"
+)
+
+var (
+	errNilArgument = errors.New("empty argument")
+	errUnsupported = errors.New("unsupported structure version")
+)
 
 // Put saves passed container structure in NeoFS system
 // through Container contract call.
 //
 // Returns calculated container identifier and any error
 // encountered that caused the saving to interrupt.
-func (w *Wrapper) Put(cnr *Container) (*CID, error) {
-	panic("implement me")
+func (w *Wrapper) Put(cnr *container.Container, pubKey, signature []byte) (*container.ID, error) {
+	if cnr == nil || len(pubKey) == 0 || len(signature) == 0 {
+		return nil, errNilArgument
+	}
+
+	args := client.PutArgs{}
+	args.SetPublicKey(pubKey)
+	args.SetSignature(signature)
+
+	id := new(container.ID)
+
+	if v2 := cnr.ToV2(); v2 == nil {
+		return nil, errUnsupported // use other major version if there any
+	} else {
+		data, err := v2.StableMarshal(nil)
+		if err != nil {
+			return nil, errors.Wrap(err, "can't marshal container")
+		}
+
+		id.SetSHA256(sha256.Sum256(data))
+		args.SetContainer(data)
+	}
+
+	return id, w.client.Put(args)
 }
 
 // Get reads the container from NeoFS system by identifier
 // through Container contract call.
 //
 // If an empty slice is returned for the requested identifier,
 // storage.ErrNotFound error is returned.
-func (w *Wrapper) Get(cid CID) (*Container, error) {
-	panic("implement me")
+func (w *Wrapper) Get(cid *container.ID) (*container.Container, error) {
+	if cid == nil {
+		return nil, errNilArgument
+	}
+
+	args := client.GetArgs{}
+
+	if v2 := cid.ToV2(); v2 == nil {
+		return nil, errUnsupported // use other major version if there any
+	} else {
+		args.SetCID(v2.GetValue())
+	}
+
+	// ask RPC neo node to get serialized container
+	rpcAnswer, err := w.client.Get(args)
+	if err != nil {
+		return nil, core.ErrNotFound
```

Comment:
I wasn't sure about this, because contract returns `FAULT` state if container does not exists. So technically `ErrNotFound` here is correct here. Length check down there is more like double check, that can be omitted, but I just ported it anyway.

Reply:
So I would rather get rid of length check.

Reply:
@alexvanin, lets get rid of slice emptiness check. At the same time, lets not lose `err`:
`return nil, errors.Wrap(core.ErrNotFound, err.Error())` or at least `TODO: log err`

Reply:
@cthulhu-rider I think `errors.Wrap(core.ErrNotFound, err.Error())` is quite good there. Fixed.